tests/driver/runner_test.py

```diff
@@ -4748,6 +4748,309 @@ async def test_delete_async_search(self, es):
+        r = runner.ClosePointInTime()
+        async with runner.CompositeContext():
+            runner.CompositeContext.put("open-pit-task1", pit_id)
+            await r(es, params)
+
+        es.close_point_in_time.assert_called_once_with(body={"id": "0123456789abcdef"}, params={}, headers=None)
+
+
+class QueryWithSearchAfterScrollTests(TestCase):
+    @mock.patch("elasticsearch.Elasticsearch")
+    @run_async
+    async def test_search_after_with_pit(self, es):
+        pit_op = "open-point-in-time1"
+        pit_id = "0123456789abcdef"
+        params = {
+            "name": "search-with-pit",
+            "index": "test-index",
+            "operation-type": "paginated-search",
+            "with-point-in-time-from": pit_op,
+            "pages": "all",
+            "results-per-page": 2,
+            "body": {
+                "sort": [{"timestamp": "asc", "tie_breaker_id": "asc"}],
+                "query": {"match-all": {}}
+            }
+        }
+
+        page_1 = {
+            "pit_id": "fedcba9876543210",
+            "took": 10,
+            "timed_out": False,
+            "hits": {
+                "total": {
+                    "value": 3,
+                    "relation": "eq"
+                },
+                "hits": [
+                    {
+                        "_id": "1",
+                         "timestamp": 1609780186,
+                         "sort": [1609780186, "1"]
+                    },
+                    {
+                        "_id": "2",
+                         "timestamp": 1609780186,
+                         "sort": [1609780186, "2"]
+                    }
+                ]
+            }
+        }
+
+        page_2 = {"pit_id": "fedcba9876543211",
+                 "took": 10,
+                 "timed_out": False,
+                 "hits": {
+                     "total": {
+                         "value": "3",
+                         "relation": "eq"
+                     },
+                     "hits": [
+                         {"_id": "3",
+                          "timestamp": 1609780187,
+                          "sort": [1609780187, "3"]
+                          }
+                     ]
+                 }}
+
+        es.transport.perform_request.side_effect = [as_future(io.BytesIO(json.dumps(page_1).encode())),
+                                                    as_future(io.BytesIO(json.dumps(page_2).encode()))]
+
+        r = runner.Query()
+
+        async with runner.CompositeContext():
+            runner.CompositeContext.put(pit_op, pit_id)
+            await r(es, params)
+            # make sure pit_id is updated afterward
+            self.assertEqual("fedcba9876543211", runner.CompositeContext.get(pit_op))
+
+        es.transport.perform_request.assert_has_calls([mock.call('GET', '/_search', params={},
+                                                                 body={'query': {'match-all': {}},
```

Comment:
Nit: Personally, I find the following formatting easier to read:

```python
 body={
     'query': {
         'match-all': {}
     },
       'sort': [{
           'timestamp': 'asc',
           'tie_breaker_id': 'asc'
       }],
       'size': 2,
       'pit': {
           'id': '0123456789abcdef',
           'keep_alive': '1m'
       }
 },
```

I'm fine if we keep it as is but I want to offer it as a thought.